lightning/src/chain/onchaintx.rs

```diff
@@ -382,7 +382,36 @@ impl<ChannelSigner: Sign> OnchainTxHandler<ChannelSigner> {
 		where F::Target: FeeEstimator,
 					L::Target: Logger,
 	{
-		if cached_request.outpoints().len() == 0 { return None } // But don't prune pending claiming request yet, we may have to resurrect HTLCs
+		let request_outpoints = cached_request.outpoints();
+		if request_outpoints.is_empty() {
+			// Don't prune pending claiming request yet, we may have to resurrect HTLCs
+			return None;
```

Comment:
If the package is `Untractable`, I think this case is pathological ? Like for the commitment transaction, we cannot aggregate them, so all the inputs (here only one) are claimed and we discard the claim request. If I'm correct we can log_error or assert that case.